core/trace.go

```diff
@@ -71,6 +78,23 @@ func CreateTraceEnv(chainConfig *params.ChainConfig, chainContext ChainContext,
 		}
 	}
 
+	// Collect start queue index, we should always have this value for blocks
+	// that have been executed.
+	// FIXME: This value will be incorrect on the signer, since we reuse this
+	// DB entry to signal which index the worker should continue from.
+	// Example: Ledger A <-- B <-- C. Block `A` contains up to `QueueIndex=9`.
+	// For block `B`, the worker skips 10 messages and includes 0.
+	// `ReadFirstQueueIndexNotInL2Block(B)` will then return `20` on the
+	// signer to avoid re-processing the same 10 transactions again for
+	// block `C`.
+	// `ReadFirstQueueIndexNotInL1Block(B)` will return the correct value
+	// `10` on follower nodes.
+	startL1QueueIndex := rawdb.ReadFirstQueueIndexNotInL2Block(chaindb, parent.Hash())
```

Comment:
I have a suggestion to fix the different behaviors between signer and follower node. In the signer node, it should also write the `startL1QueueIndex` from the parent block + `numL1Messages` in this block to the `FirstQueueIndexNotInL1Block` db entry. The `nextL1MsgIndex` should be kept locally in the memory to avoid redundant work.

Reply:
per offline disscussion with @Thegaram , we leave this as a todo and will fix it in a follow up PR.